lib/crypto/ecdsa.js

```diff
@@ -88,9 +88,10 @@ ECDSA.prototype.deterministicK = function(badrs) {
   var x = this.privkey.bn.toBuffer({
     size: 32
   });
-  k = Hash.sha256hmac(Buffer.concat([v, new Buffer([0x00]), x, this.hashbuf]), k);
+  var hashbuf = this.endian === 'little' ? [].reverse.call(new Buffer(this.hashbuf)) : this.hashbuf
```

Comment:
We may want to use `BufferUtil.reverse()` here, since we have benchmarks around it, and use it elsewhere.